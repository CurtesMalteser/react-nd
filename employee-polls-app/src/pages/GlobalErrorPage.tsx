
function GlobalErrorPage() {
    return (
        <h1>Not Found!</h1>
    );
}

export default GlobalErrorPage;