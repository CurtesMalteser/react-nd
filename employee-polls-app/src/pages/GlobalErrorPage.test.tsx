import { cleanup, render } from '@testing-library/react';
import GlobalErrorPage from './GlobalErrorPage';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { AnyAction, Store, configureStore } from '@reduxjs/toolkit';
import getDefaultState from '../features/testUtils';

describe('GlobalErrorPage', () => {

    let _store: Store<unknown, AnyAction> | null;

    let store = () => {
        if (_store === null) {
            throw new Error("Store has not been initialized");
        }
        return _store;
    };

    beforeEach(() => {
        _store = configureStore({
            reducer: getDefaultState,
        }
        );
    });

    afterEach(() => {
        _store = null;
        cleanup();
    });

    it('should match the snapshot', () => {
        const { asFragment } = render(
            <Provider store={store()}>
                <Router>
                    <GlobalErrorPage />
                </Router>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});