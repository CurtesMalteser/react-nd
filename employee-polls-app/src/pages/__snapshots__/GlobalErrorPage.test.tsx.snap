// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalErrorPage should match the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-body-tertiary navbar navbar-expand-lg navbar-light"
    style="margin-bottom: 48px;"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Employee Polls
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link"
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <a
            class="nav-link"
            href="/add"
          >
            New
          </a>
        </div>
        <div
          class="d-flex"
          style="margin-right: 24px;"
        >
          <img
            alt="Profile Picture undefined"
            class="rounded-circle"
            src="employees_pool_logo.jpg"
            style="width: 50px; height: 50px; margin-right: 10px;"
          />
        </div>
        <button
          class="btn btn-success"
          type="button"
        >
          Login
        </button>
      </div>
    </div>
  </nav>
  <div
    class="md-6 container"
    style="margin: 20px auto 20px auto;"
  >
    <div
      class="col"
    >
      <div
        class="d-flex justify-content-center row"
      >
        <img
          alt="logo"
          class="rounded-circle"
          src="employees_pool_logo.jpg"
          style="width: 400px; height: 400px; margin-bottom: 48px;"
        />
      </div>
      <div
        class="row"
        style="margin-bottom: 48px;"
      >
        <h1
          class="d-flex justify-content-center"
        >
          Oops! Somenthing went wrong...
        </h1>
      </div>
      <div
        class="row"
        style="margin-bottom: 48px;"
      >
        <h2
          class="d-flex justify-content-center"
        >
          404: We coudn't find the path you're looking for.
        </h2>
      </div>
    </div>
  </div>
  <div
    style="display: flex; justify-content: center;"
  >
    <button
      class="btn btn-success"
      type="button"
    >
      <svg
        class="bi bi-house-up-fill"
        fill="currentColor"
        height="24"
        style="margin-right: 10px;"
        viewBox="0 0 16 16"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.5 16a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7m.354-5.854 1.5 1.5a.5.5 0 0 1-.708.708L13 11.707V14.5a.5.5 0 1 1-1 0v-2.793l-.646.647a.5.5 0 0 1-.708-.707l1.5-1.5a.5.5 0 0 1 .708 0Z"
        />
        <path
          d="M8.707 1.5a1 1 0 0 0-1.414 0L.646 8.146a.5.5 0 0 0 .708.708L8 2.207l6.646 6.647a.5.5 0 0 0 .708-.708L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293z"
        />
        <path
          d="m8 3.293 4.712 4.712A4.5 4.5 0 0 0 8.758 15H3.5A1.5 1.5 0 0 1 2 13.5V9.293z"
        />
      </svg>
      Go Home
    </button>
  </div>
</DocumentFragment>
`;
